data_processing/meteo/task_functions.py

```diff
@@ -0,0 +1,171 @@
+from datagouvfr_data_pipelines.config import (
+    AIRFLOW_DAG_HOME,
+    AIRFLOW_DAG_TMP,
+    DATAGOUV_SECRET_API_KEY,
+    AIRFLOW_ENV,
+    MINIO_URL,
+    SECRET_MINIO_DATA_PIPELINE_USER,
+    SECRET_MINIO_DATA_PIPELINE_PASSWORD,
+)
+from datagouvfr_data_pipelines.utils.datagouv import (
+    post_remote_resource,
+    DATAGOUV_URL
+)
+from datagouvfr_data_pipelines.utils.mattermost import send_message
+from datagouvfr_data_pipelines.utils.minio import send_files, get_all_files_from_parent_folder
+import ftplib
+import os
+import json
+from datetime import datetime
+import re
+import requests
+
+DAG_FOLDER = "datagouvfr_data_pipelines/data_processing/"
+DATADIR = f"{AIRFLOW_DAG_TMP}meteo/data"
+
+
+def list_ftp_files_recursive(ftp, path='', base_path=''):
+    files = []
+    try:
+        ftp.cwd(path)
+        current_path = f"{base_path}/{path}" if base_path else path
+        ftp.retrlines('LIST', lambda x: files.append((current_path.split('//')[-1], x.split()[-1])))
+        for item in files:
+            if '.' not in item[1]:
+                files += list_ftp_files_recursive(ftp, f"{path}/{item[1]}", current_path)
+    except ftplib.error_perm:
+        pass
+    return files
+
+
+def get_current_files_on_ftp(ti, ftp):
+    ftp_files = list_ftp_files_recursive(ftp)
+    print(ftp_files)
+    ftp_files = [
+        path + '/' + file for path, file in ftp_files if '.' in file
```

Comment:
I'm not sure what this is for

Reply:
The function `list_ftp_files_recursive` returns a list of tuples `(file_path, file_name)`, so I just reprocess it to have a simple list of `file_path/file_name`. In the next iteration it will return either the file size or a file checksum too (for comparison with Minio)